packages/medusa/src/services/product.ts

```diff
@@ -833,7 +834,7 @@ class ProductService extends TransactionBaseService {
         )
       }
 
-      const result = await this.retrieve(productId)
+      const result = await this.withTransaction(manager).retrieve(productId)
```

Comment:
suggestion: same as the previous one to not recreate the instance multiple times everytime